[TASK] Refactor Creator model

calien666 opened the pull request:
This PR refactors the Creator model used for Creator jobs to work without Extbase.

calien666 committed:
[TASK] refactor Creator

This commit refactors the Creator class to get rid of extbase usage

calien666 committed:
[TASK] Update to main

This commit adds the new isRunning to changed repository

calien666 committed:
[TASK] Remove PersistenceManager

Removes the persistenceManager from RunCommand as no longer needed

calien666 committed:
[TASK] Increase minimum PHP version

As Creator uses constructor attributes, the minimum PHP level needs to be 8.1

calien666 committed:
[TASK] Add movable Creator dataset

This commit adds the DataHandler logic for moving the Creator dataset to the newly created page

calien666 committed:
[BUGFIX] Add missing parameters

sbuerk commented on Classes/Domain/Repository/CreatorRepository.php:
```diff
@@ -13,42 +13,156 @@
+    public function __construct(
+        ConnectionPool $connection
+    ) {
+        $this->queryBuilder = $connection->getQueryBuilderForTable(self::$table);
```
Don't store a QueryBuilder instance directly as property. QueryBuilder instances must not be reused, and per design changed on things. So either it needs to be ensured that this is not used directly but instead cloned to a variable before usage.

Better save the connection here and use $this->connection->createQueryBuilder()` in the methods instead.

sbuerk commented on composer.json:
```diff
@@ -26,7 +26,7 @@
 	"require": {
 		"ext-curl": "*",
-		"php": "8.0.*||8.1.*||8.2.*",
+		"php": "8.1.* || 8.2.*",
```
Why is php 8.3 omitted hear ? (already before)

I would suggest to use

```
  "php": "^8.1 || ^8.2"
```

here. In the wild, such a contraint is used as

* support php 8.1 and 8.2
* allow php 8.3 for testing/execution without offical support and without using unsecure `--ignore-platfrom-req`

Not sure about the policy at all - from a personal point of view all php version allowed by supported core version should be allowed/supported by a extension - which would mean php7.4 for a v11 support. Using newer php language feature should be ommited.

As this is already broken, not sure about the min version raise - eventually leave out
the "readonly" for the CPP and keep php8.0 (CPP is possible since 8.0, readonly since 8.1). Not sure about overall version policy of this extension.

calien666 committed:
[TASK] decrease used PHP version to 8.0

And add setters who are needed to Creator. Refactor repository for
instantiating QueryBuilder per use instead of class usage
